### PR TITLE
Update logged in view with extension workshop

### LINKED ISSUE #10910: Update logged-in /developers/ with links to extensions-workshop
Ux to define link copy and placement:

Currently looks like this for reference:
Developer_Hub____Add-ons_for_Firefox

### PR DESCRIPTION
**Pull request by championshuttler**
Fixes #10910

### CODE REVIEW

File: src/olympia/devhub/templates/devhub/new-landing/components/navigation.html [hunk 1]
```diff
@@ -11,6 +11,9 @@
         <li class="show-on-desktop"><a href="{{ url('devhub.addons') }}">
           {{ _('My Add-ons') }}
         </a></li>
+        <li><a href="https://extensionworkshop.com/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=devhub-menu-link">
```
Comment (championshuttler) [inline src/olympia/devhub/templates/devhub/new-landing/components/navigation.html]: If I am not wrong, we are going to show it everywhere so did not add `show on desktop` here.
Correct me if i am wrong.

Thank you

Comment (eviljeff) [reply src/olympia/devhub/templates/devhub/new-landing/components/navigation.html]: I guess?
@caitmuenster do we want the link shown on mobile-sized views or just when viewed on a larger screen? (Not sure there's enough space to fit it on mobile tbh, unless we drop one of the other links)

Comment (caitmuenster) [reply src/olympia/devhub/templates/devhub/new-landing/components/navigation.html]: We can drop the link on mobile. :) Thanks for checking in!

Comment (championshuttler) [reply src/olympia/devhub/templates/devhub/new-landing/components/navigation.html]: Thanks for clarification @caitmuenster!

File: src/olympia/devhub/templates/devhub/new-landing/components/navigation.html [hunk 2]
```diff
@@ -11,6 +11,9 @@
         <li class="show-on-desktop"><a href="{{ url('devhub.addons') }}">
           {{ _('My Add-ons') }}
         </a></li>
+        <li class="show-on-desktop"><a href="https://extensionworkshop.com/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=devhub-menu-link">
+          {{ _('Extension Workshop') }}
+        </a></li>
```
Comment (eviljeff) [inline src/olympia/devhub/templates/devhub/new-landing/components/navigation.html]: lets move this out of the `{% if ... %}` block so it's shown to logged out users too.

File: src/olympia/devhub/templates/devhub/new-landing/components/navigation.html [hunk 3]
```diff
@@ -13,6 +13,9 @@
         </a></li>
       {% endif %}
 
+      <li class="show-on-desktop"><a href="https://extensionworkshop.com/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=devhub-menu-link">
+          {{ _('Extension Workshop') }}
```
Comment (championshuttler) [inline src/olympia/devhub/templates/devhub/new-landing/components/navigation.html]: @eviljeff I made the changes, but as per bug description (maybe it needs to be updated), we might need to update the logged in view. If not lets 🚢 it.

Thanks

**Pull request merged**